tools/generator-go-sdk/generator/templater_methods_autorest.go

```diff
@@ -653,6 +652,66 @@ func (c %[1]s) responderFor%[2]s(resp *http.Response) (result %[6]s, err error)
 		return &output, nil
 	}
 
+	if discriminatedType != "" && strings.HasPrefix(c.operationName, "List") {
```

Comment:
not every list operation starts with `List` unfortunately, we'd be better to do:

```suggestion
	if discriminatedType != "" && c.operation.FieldContainingPaginationDetails != nil {
```